docs/guide/in-depth/advanced-chunks.md

```diff
@@ -1,6 +1,238 @@
+- `hash0` is the hash of the output file, it changes if the content of the file changes.
+
+### Reduce cache invalidation
+
+Let's talk about cache invalidation first. Cache invalidation here means that when you deploy a new version of your application, the browser will need to download the new version of the file. If the file is large, this can lead to a poor user experience.
+
+For example, you change the `app.jsx` file
+
+```jsx [app.jsx]
+function App() {
+  return <Button onClick={() => alert('Button clicked!')} />; // [!code --]
+  return <Button onClick={() => alert('Button clicked!!!')} />; // [!code ++]
+}
+```
+
+and naturally, you get a `output-hash1.js` file with the same content as `output-hash0.js`, except the `App` function is changed.
+
+Now, if you deploy this new version of your application, the browser will need to download the entire `output-hash1.js` file, even though only a small part of it has changed. This is because the hash of the file has changed, and the browser will treat it as a new file.
+
+To solve this problem, we can use `advancedChunks` to split output libraries into separate chunks, because they don't change frequently.
+
+```js [rolldown.config.js]
+export default {
+  // ... other configurations
+  advancedChunks: {
+    groups: [
+      {
+        test: /node_modules/,
+        name: 'libs',
+      },
+    ],
+  },
+};
+```
+
+By using the above `advancedChunks` option, the output will look like this:
+
+:::code-group
+
+```js [output-hash0.js]
+import ... from './libs-hash0.js';
+// App.js
+function App() {
+  return <Button onClick={() => alert("Button clicked!")} />;
+}
+
+// index.js
+
+ReactDom.createRoot(document.getElementById("root")).render(<App />);
+```
+
+```js [libs-hash0.js]
+// node_modules/react/index.js
+"React library code";
+
+// node_modules/ui-lib/index.js
+"UI library code";
+
+// node_modules/react-dom/index.js
+"ReactDOM library code";
+
+export { ... };
+```
+
+:::
+
+For example, after you change the `app.jsx` file
+
+```jsx [app.jsx]
+function App() {
+  return <Button onClick={() => alert('Button clicked!')} />; // [!code --]
+  return <Button onClick={() => alert('Button clicked!!!')} />; // [!code ++]
+}
+```
+
+you will get output like this:
+
+:::code-group
+
+```js [output-hash1.js]
+import ... from './libs-hash0.js';
+// App.js
+function App() {
+  return <Button onClick={() => alert("Button clicked!!!")} />;
+}
+
+// index.js
+
+ReactDom.createRoot(document.getElementById("root")).render(<App />);
+```
+
+```js [libs-hash0.js]
+// node_modules/react/index.js
+"React library code";
+
+// node_modules/ui-lib/index.js
+"UI library code";
+
+// node_modules/react-dom/index.js
+"ReactDOM library code";
+
+export { ... };
+```
+
+:::
+
+- The `libs-hash0.js` file is not changed, so the browser can use the cached version of the file.
+- The `output-hash1.js` file is changed, so the browser will download the new version of the file.
+
```

Comment:
It may be worth adding a note that using a function that was tree-shaken in the previous bundle will invalidate the library chunk.

Reply:
Yes. Real-world case might be complicated. I'll add warning here.